Clean up and format markdown documentation

jacobherrington opened the pull request:
### Description
I am reading through most of Puma's documentation, so I decided to make some minor improvements!

Generally, I am trying to reduce wordiness, improve clarity, and fix mechanical mistakes.

Additionally, in this PR I used my text editor to reformat some of the files. It generates a *lot* of git noise, so I only did that on files that seemed to be word-wrapped at one time in the past. I think this is useful for a couple of reasons, 1) it makes the git history of these files much more useful and clear going forward (changing a comma doesn't result in a diff that touches an entire paragraph this way), 2) some people do prefer to read these files in tmux panes or tiled text editors and I don't mind hitting a hotkey to accommodate them :).

If these changes are too broad or not deemed useful enough, I can always undo the formatting step.

### Your checklist for this pull request
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have reviewed the [guidelines for contributing](../blob/master/CONTRIBUTING.md) to this repository.
- [x] I have added (or updated) appropriate tests if this PR fixes a bug or adds a feature.
- [ ] *My pull request is 100 lines added/removed or less so that it can be easily reviewed.*  🙈🙈🙈
- [x] If this PR doesn't need tests (docs change), I added `[ci skip]` to the title of the PR.
- [x] If this closes any issues, I have added "Closes `#issue`" to the PR description or my commit messages.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed, including Rubocop.

nateberkopec commented on docs/architecture.md:
```diff
@@ -4,38 +4,71 @@
 ![https://bit.ly/2iJuFky](images/puma-general-arch.png)
 
-Puma is a threaded Ruby HTTP application server, processing requests across a TCP or UNIX socket.
+Puma is a threaded Ruby HTTP application server processing requests across a TCP
```
`that processes requests on TCP and/or UNIX sockets` would be most accurate

nateberkopec commented on docs/deployment.md:
```diff
@@ -48,54 +45,66 @@ See [systemd.md](systemd.md)
-puma, described separately in our [restart documentation](restart.md).
+Puma to start running that new code. There are a few options for restarting
+Puma, described separately in our [restart documentation](restart.md).
+# Deployment engineering for Puma
```
Mistake here?

nateberkopec commented:
Looks good except for the one possible copy paste error I found.

One thing I've always thought about re: our docs was could we make them easier for ESL speakers to understand - decrease reading level, simplify vocabulary, etc. Thought for the future.

jacobherrington commented:
@nateberkopec Thanks for the review. I addressed your comments and pushed new commits.

I agree with the goal of simplifying the language, especially for people who aren't native English speakers or aren't familiar with idioms that might be present in the docs. 👍
